### PR TITLE
feat: arconnect components v2

### PR DESCRIPTION
**Pull request by elehma4**
This PR integrates v2 components into ArConnect replacing all existing components. 

It includes new components for:
- Buttons
- Inputs
- List Items
- Selects
- Tooltips
- Modal Popups

### CODE REVIEW

**Review by nicholaswma — CHANGES_REQUESTED** [review]
bg color for search bar in currency needs to be updated

**Review by 7i7o — APPROVED** [review]
tACK 🚀

File: src/components/dashboard/Contacts.tsx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { Spacer, useInput } from "@arconnect/components";
+import { ButtonV2, Spacer, useInput } from "@arconnect/components";
```
Comment (nicholaswma) [inline src/components/dashboard/Contacts.tsx:1]: couldn't comment it out but just remove 
`import { IconButton } from "~components/IconButton";`

File: src/components/dashboard/Wallets.tsx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { concatGatewayURL } from "~gateways/utils";
-import { Spacer, useInput } from "@arconnect/components";
+import { ButtonV2, Spacer, useInput } from "@arconnect/components";
```
Comment (nicholaswma) [inline src/components/dashboard/Wallets.tsx:2]: remove:
import { PlusIcon } from "@iconicicons/react";
import { IconButton } from "~components/IconButton";

File: src/components/dashboard/list/WalletListItem.tsx [hunk 3]
```diff
@@ -36,19 +36,19 @@ export default function WalletListItem({
       dragListener={false}
       dragControls={dragControls}
     >
-      <BaseElement
+      <ListItem
```
Comment (nicholaswma) [inline src/components/dashboard/list/WalletListItem.tsx:39]: type error on dragcontrols - I think this is probably caused by the mismatch in framer versions, we can investigate later on

File: src/components/dashboard/Contacts.tsx [hunk 4]
```diff
@@ -200,21 +200,18 @@ const LetterHeader = styled.div`
...
   top: 0;
   left: 0;
   right: 0;
   z-index: 20;
   grid-template-columns: auto auto;
+  background-color: rgb(${(props) => props.theme.cardBackground});
 `;
 
-const AddContactButton = styled(IconButton).attrs({
+const AddContactButton = styled(ButtonV2).attrs({
```
Comment (nicholaswma) [inline src/components/dashboard/Contacts.tsx:211]: Should this be primary?

File: src/components/dashboard/Wallets.tsx [hunk 5]
```diff
@@ -156,22 +156,18 @@ const Wrapper = styled.div`
...
   top: 0;
   left: 0;
   right: 0;
   z-index: 20;
   grid-template-columns: auto auto;
+  background-color: rgb(${(props) => props.theme.cardBackground});
 `;
 
-const AddWalletButton = styled(IconButton).attrs({
+const AddWalletButton = styled(ButtonV2).attrs({
```
Comment (nicholaswma) [inline src/components/dashboard/Wallets.tsx:167]: Should this be primary?

File: src/components/dashboard/list/TokenListItem.tsx [hunk 6]
```diff
@@ -2,17 +2,18 @@ import { DREContract, DRENode, NODES } from "@arconnect/warp-dre";
...
 import { useTheme } from "~utils/theme";
 import { useLocation } from "wouter";
 import * as viewblock from "~lib/viewblock";
-import BaseElement from "./BaseElement";
 import styled from "styled-components";
 import { useGateway } from "~gateways/wayfinder";
 import { concatGatewayURL } from "~gateways/utils";
 import aoLogo from "url:/assets/ecosystem/ao-logo.svg";
 import { getUserAvatar } from "~lib/avatar";
+import BaseElement from "./BaseElement";
```
Comment (7i7o) [inline src/components/dashboard/list/TokenListItem.tsx]: Wasn't BaseElement replaced by ListItem?
We shouldn't be using it for the AO tokens

File: src/components/dashboard/Contacts.tsx (hunk 1 above)
Comment (7i7o) [inline src/components/dashboard/Contacts.tsx:1]: `ButtonV2` replaced `IconButtton`, so the `IconButton` import should be removed

File: src/components/dashboard/Reset.tsx [hunk 7]
```diff
@@ -83,15 +92,19 @@ const Warning = styled(Text)`
...
-const ResetButton = styled(Button).attrs({
+const ResetButton = styled(ButtonV2).attrs({
   secondary: true,
   fullWidth: true
-})`
-  background-color: rgba(255, 0, 0, 0.2);
-  color: #ff0000;
+})<{ displayTheme: DisplayTheme }>`
+  background-color: ${(props) =>
+    props.displayTheme === "light" ? "#F58080" : "#8C1A1A"};
```
Comment (7i7o) [inline src/components/dashboard/Reset.tsx]: Same thing Nick mentioned, we should get some clarity on what color to use from the theme here so we don't hardcode the colors

File: src/components/dashboard/subsettings/AddWallet.tsx [hunk 8]
```diff
@@ -11,7 +11,9 @@ import {
   Text,
```
Comment (7i7o) [inline src/components/dashboard/subsettings/AddWallet.tsx:9]: Still importing `Button` and `Input` in the previous 2 lines

File: src/components/dashboard/subsettings/AppSettings.tsx [hunk 9]
```diff
@@ -243,13 +249,13 @@ export default function AppSettings({ app, showTitle = false }: Props) {
         <option value="custom" selected={isCustom}>
           Custom
         </option>
-      </Select>
+      </SelectV2>
       {editingCustom && (
         <>
           <Spacer y={0.8} />
           <InputWithBtn>
```
Comment (7i7o) [inline src/components/dashboard/subsettings/AppSettings.tsx:256]: `InputWithBtn` component in the `arlocal` folder (and some other files in that folder) still use `Button` and `Input`

File: src/components/dashboard/subsettings/AppSettings.tsx [hunk 10]
```diff
@@ -243,13 +249,13 @@ export default function AppSettings({ app, showTitle = false }: Props) {
...
             <InputWrapper>
-              <Input
+              <InputV2
```
Comment (7i7o) [inline src/components/dashboard/subsettings/AppSettings.tsx:258]: Inside this component we are using `IconButton` that still uses `Button`, as well as `Recipient.tsx` and `SliderMenu.tsx` in the components folder

File: src/components/popup/WalletHeader.tsx [hunk 11]
```diff
@@ -380,6 +400,46 @@ export default function WalletHeader() {
         />
       )}
       <WalletSwitcher open={isOpen} close={() => setOpen(false)} />
+      {/* @ts-ignore */}
```
Comment (7i7o) [inline src/components/popup/WalletHeader.tsx]: Why are we ignoring typescript errors?

File: src/routes/auth/connect.tsx [hunk 12]
```diff
@@ -299,10 +300,10 @@ const WalletSelectWrapper = styled.div`
 `;
 
 const SelectIcon = styled(ChevronDownIcon)`
-  font-size: 1.25rem;
-  width: 1em;
-  height: 1em;
-  color: rgb(${(props) => props.theme.theme});
+  font-size: 16px;
+  width: 22px;
+  height: 22px;
```
Comment (7i7o) [inline src/routes/auth/connect.tsx]: Let's user `rem` units for a11y

File: src/routes/auth/connect.tsx [hunk 13]
```diff
@@ -313,25 +314,34 @@ const WalletSelect = styled(Card)<{ open: boolean }>`
   justify-content: space-between;
   cursor: pointer;
   background-color: transparent;
-  padding: 0.75rem 1.25rem;
-  border-radius: 18px;
+  padding: 13.5px 15px;
```
Comment (7i7o) [inline src/routes/auth/connect.tsx]: Same with `rem` units

File: src/routes/popup/send/index.tsx [hunk 14]
```diff
@@ -3,6 +3,7 @@ import { useState, useEffect, useMemo } from "react";
 import styled, { css } from "styled-components";
 import {
   Button,
+  ButtonV2,
   Input,
```
Comment (7i7o) [inline src/routes/popup/send/index.tsx]: `Button` and `Input` still in use, `ButtonV2` imported but not used?

**Review by 7i7o — COMMENTED** [review]
Left comments in each file.

AO Tokens in settings are still using BaseElement, so thay don't look the same as Warp tokens

**Review by nicholaswma — APPROVED** [review]

**Review by 7i7o — APPROVED** [review]
LGTM 🚀

**Pull request merged**